rtl/eclrtl/rtlformat.hpp

```diff
@@ -256,6 +258,7 @@ class ECLRTL_API CommonJsonWriter : implements IXmlWriterExt, public CInterface
 
     void outputBeginRoot(){out.append('{');}
     void outputEndRoot(){out.append('}');}
+    virtual void flushContent(bool close) override { flush(close); }
```

Comment:
@wangkx did you test this for JSON?  JSON is a little more state aware than XML.  XML is a series of <> and </> but JSON needs things like commas in the right places depending on what came before.  Having the outside tell us exactly when to flush might mean we need to remember something like the last character.  I didn't look at that code to see, but if you tested it streaming out a large result as json that would show if it got the commas, etc in the right places.

Reply:
I did test this for JSON (data size about 6.6k bytes, single column length from 1 to 200, and Flush Threshold 60 bytes). The JSON output seems valid. I guess that JSON writer does not need to read previous output in order to decide current comma or quote. Does the writer already remember that? If not, my test should not generate a valid JSON format.

Reply:
Ok, yes you are correct, I checked the JSONWriter code and it does keep track of needed delimiters a different way.  Thanks for testing, looks good.